data-generation/src/main/scala/thesis/LandyConsumer.scala

```diff
@@ -0,0 +1,72 @@
+package thesis
+
+import org.apache.spark.SparkContext
+import org.apache.spark.graphx.{Edge, Graph}
+import org.apache.spark.rdd.RDD
+import thesis.Action.{CREATE, UPDATE}
+import thesis.LTSV.Attributes
+import utils.LogUtils
+
+import scala.collection.mutable
+
+object LandyConsumer {
+
+  def createEdge(log: LogTSV, validTo: String): Edge[Attributes] = {
+    log.entity match {
+      case Entity.VERTEX(_) => throw new IllegalStateException("This should not be called on a log with vertices")
+      case Entity.EDGE(srcId, dstId) => {
+        val attributesWithValidTime = log.attributes.+("validFrom"->log.timestamp.toString, "validTo"->validTo)
+        Edge(srcId, dstId, attributesWithValidTime)
+      }
+    }
+  }
+
+  def createVertex(log: LogTSV, validTo: String): (Long, Attributes) = {
+    log.entity match {
+      case Entity.VERTEX(objId) => {
+        if (log.timestamp.toString == null || validTo == null) println("this aint good")
+        val attributesWithValidTime = log.attributes.+("validFrom" -> log.timestamp.toString, "validTo" -> validTo)
+        (objId, attributesWithValidTime)
+      }
+      case Entity.EDGE(_, _) => throw new IllegalStateException("This should not be called on a log with edges")
+    }
+  }
+
+  def generateEdges(logs: Iterable[LogTSV]): Iterable[Edge[Attributes]] = {
+    val backwardOrderedLogs = logs.toSeq.sortWith((log1, log2) => log1.timestamp.isBefore(log2.timestamp))
+    val aVeryBigDate = "9999-01-01T00:00:00.000Z"
+    var lastTimestamp = aVeryBigDate
+    val edges = mutable.MutableList[Edge[Attributes]]()
+    for (log <- backwardOrderedLogs) {
+      if (log.action == CREATE || log.action == UPDATE) {
```

Comment:
hva skjer med DELETE?

Reply:
Er liiiiitt usikker på om det her blir rett. Men vi starter baklengs for å få satt validTo på en bedre måte, sant. Og hvis det er en DELETE som er sist, så vil validTo på den siste UPDATE-en være tidsstempelet til DELETE-en. Så når det er DELETE ønsker vi bare å finne ut av slettetidspunktet

Reply:
Ahh, tror jeg skjønner. Ble bare litt vanskelig å lese når jeg ikke så på en måte at DELETE var håndtert

Reply:
Legger inn en kommentar på det!